kpi/management/commands/provision_server.py

```diff
@@ -0,0 +1,186 @@
+        return parser
+
+    def add_arguments(self, parser):
+        subparsers = parser.add_subparsers(dest='command')
+
+        # Subcommand for managing social apps
+        socialapp_parser = subparsers.add_parser(
+            'socialapp', help='Insert a social app and related custom data'
+        )
+
+        socialapp_parser.add_argument(
+            '--provider',
+            type=str,
+            required=True,
+            help='Either openid_connect or microsoft',
+        )
+        socialapp_parser.add_argument(
+            '--provider_id',
+            type=str,
+            required=True,
+            help='Provider ID used in the login url',
+        )
+        socialapp_parser.add_argument(
+            '--name',
+            type=str,
+            required=True,
+            help='Name of the organization displayed on the login page',
+        )
+        socialapp_parser.add_argument(
+            '--client_id',
+            type=str,
+            required=True,
+            help='App ID, or consumer key',
+        )
+        socialapp_parser.add_argument(
+            '--secret',
+            type=str,
+            default='',
+            help='API secret, client secret, or consumer secret',
+        )
+        socialapp_parser.add_argument(
+            '--key',
+            type=str,
+            default='',
+            help='Key provided by client',
+        )
+        socialapp_parser.add_argument(
+            'settings',
+            nargs='?',
+            type=str,
+            default='{}',
+            help='Settings in json format enclosed with single quotes',
+        )
+
+        # Subcommand for managing constance configurations
+        config_parser = subparsers.add_parser(
+            'config', help='Update application configuration'
+        )
+        config_parser.add_argument(
+            'config_kv_pairs',
+            nargs='+',
+            type=str,
+            help='Constance configuration values as key=value pairs',
+        )
+
+    def handle(self, *args, **kwargs):
+        command = kwargs.get('command')
+
+        if command == 'socialapp':
+            self.handle_socialapp(kwargs)
+        elif command == 'config':
+            self.handle_config(kwargs)
+        else:
+            self.stdout.write(self.style.ERROR('No valid subcommand provided'))
+
+    def handle_socialapp(self, kwargs):
+        provider = kwargs['provider']
+        provider_id = kwargs['provider_id']
+        name = kwargs['name']
+        client_id = kwargs['client_id']
+        secret = os.getenv('SOCIAL_APP_SECRET') or kwargs.get('secret', '')
+        key = kwargs.get('key', '')
+        settings_json = kwargs['settings']
+
+        try:
+            settings = json.loads(settings_json)
+        except TypeError:
+            raise json.JSONDecodeError
+
+        social_app_data = {
+            'provider': provider,
+            'provider_id': provider_id,
+            'name': name,
```

Comment:
This line can be omitted, you don't need to repeat the args in get_or_create again in defaults. The usage of get_or_create makes a lot of sense though.